internal/failure_domain.go

```diff
@@ -87,6 +87,23 @@ func PickFewest(failureDomains clusterv1.FailureDomains, machines FilterableMach
 	return pointer.StringPtr(aggregations[0].id)
 }
 
+// NPickFewest returns N failure domains with the fewest number of machines.
+func NPickFewest(failureDomains clusterv1.FailureDomains, machines FilterableMachineCollection, replicas int) []string {
```

Comment:
I think this might be more consistently named as `PickFewestN` because it reads more naturally like the other similar functions.

Reply:
I would also recommend correlating the argument name with the comment/function name a little harder - the func name and doc imply there is a parameter `N`, but it's called `replicas` - just a thought. Probably sufficient to just add the word replicas to the comment?

Reply:
Yeah, I agree, it should probably be called `n` in the parameters.